src/index.ts

```diff
@@ -13,11 +13,12 @@
  */
 /* eslint-disable no-var,vars-on-top */
 // eslint-disable-next-line prefer-destructuring
-var createContext = require('react').createContext;
+import type { Context } from 'react'
+import { createContext } from 'react';
 
-var contexts = {};
+export var contexts: {[key: string]: Context<any>} = {};
 
-module.exports = function createSharedReactContext(defaultValue, key) {
+function createSharedReactContext(defaultValue: any, key: string): Context<any> {
```

Comment:
This looks like a ripe opportunity to use a generic to help eliminate the references to `any`: 

- Generics: https://www.typescriptlang.org/docs/handbook/2/generics.html
- React types for generic in `createContext` (maybe we should mirror this?): https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/v17/index.d.ts#L377